lib/unleash/feature_toggle.rb

```diff
@@ -97,10 +97,16 @@ def sum_variant_defs_weights
     end
 
     def variant_salt(context, stickiness = "default")
-      return context.get_by_name(stickiness) unless stickiness == "default"
-      return context.user_id unless context.user_id.to_s.empty?
-      return context.session_id unless context.session_id.to_s.empty?
-      return context.remote_address unless context.remote_address.to_s.empty?
+      begin
+        return context.get_by_name(stickiness) if !context.nil? && stickiness != "default"
```

Comment:
So I noticed that the previous PR passes the spec tests without this. Is this sneaky undefined behaviour we don't have codified in the spec tests?

Reply:
Correct!

You can see from the build logs in the commit at https://github.com/Unleash/unleash-client-ruby/commit/ff28b3100d5c3cb6611d2ea145531d386272ff95  (tests w/o fix) that it fails w/o the fix due to lack of relevant tests. The tests come in the following commit.

Reply:
Oof. Right, that needs to make its way into the spec tests. Not relevant to this PR. 

Looks like a sane behaviour, I support this!